Debugger.py

```diff
@@ -73,42 +74,83 @@ def parseCommandLineArgs():
                         help="This is required if -s option was used on the linker. That will allow "
                              "binary to be loader at correct address specified inside the binary")
 
+    parser.add_argument("-bp", "--breakpoint",
+                        required=False,
+                        type=str,
+                        default=None,
+                        help="This is an optional argument. If present, debugger will load breakpoints "
+                        "previously defined and stored in the specified file.")
+
     args = parser.parse_args()
 
     return args
 
 
-def validatePaths(argsWithPaths):
+def validateFilePath(filename, ext=None):
     """
-    This function will simply validate that the input path exists and that the output path
-    is free for the system to use
-    :param argsWithPaths: An input parsed object as provided by argparse.parse_args()
-    :return: This does not return. Simply raises ValueError in cases where paths are not valid.
+    This function will simply validate file existence and file extension.
+    If either is not true, then will raise an error.
+    :param filename: str, file path to be validated
+    :param ext: str, expected extension
+    :return: boolean True,
     """
-    gotSymbols = False
-    if not os.path.exists(argsWithPaths.input):
-        raise ValueError("ERROR: file {} does not exists.".format(argsWithPaths.input,))
+    if not os.path.exists(filename):
+        raise ValueError("ERROR: File {} does not exists.".format(filename,))
     else:
-        if os.path.exists(argsWithPaths.input.split(".")[0] + ".sym"):
-            gotSymbols = True
+        fileExt = filename.split(".")[-1]
+        if fileExt != ext:
+            raise ValueError("ERROR: Incorrect file extension on file {}".format(filename,))
 
-    return gotSymbols
+    return True
+
+
+def validateBreakPointFile(usable_args, ext=None):
+    """
+    This function will check the inputted breakpoint file and handle the cases.
```

Comment:
I think this comment needs to provide an example as to what is a valid break point file.

Reply:
With breakpoint files no longer needing to have a specific file extension function comment

    This function will check if user provided a breakpoint file.
    If user provided a file that does not exist, function will create the file.
    If user did not provide a file, no breakpoint file will be used.
    :param usable_args: An input parsed object as provided by argparse.parse_args()
    :return inputBreakpointFile: str, name of the breakpointFile that will be used by Debugger